sql/hive/src/test/scala/org/apache/spark/sql/hive/InsertSuite.scala

```diff
@@ -735,6 +735,39 @@ class InsertSuite extends QueryTest with TestHiveSingleton with BeforeAndAfter
     }
   }
 
+  test("Throw exception on unsafe cast with strict casting policy") {
```

Comment:
why duplicate the test here? This feature is not hive-specific and I think it's good enough to create a test in sql/core `InsertSuite`.